### PR TITLE
Add support for the !important attribute

### PR DESCRIPTION
**Pull request by LaurenzV**
This solves the problem I had where I wanted to achieve that I can override _all_ styles in the SVG via stylesheet injection. I tested it with `rsvg-convert` and it works in the same way, so it would be great to have this merged. :)

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Are you sure `!important` shouldn't be handled on the `simplecss` side? I genuinely have no idea how it suppose to work.

File: crates/usvg/src/parser/svgtree/parse.rs
Comment (RazrFalcon) [inline crates/usvg/src/parser/svgtree/parse.rs:295]: Too much copy-paste. Can we pass `declaration` directly? Or at least shorthand variables instead of typing `declaration.value, declaration.important` over and over again?

**Comment by LaurenzV** [comment]
> Are you sure !important shouldn't be handled on the simplecss side? I genuinely have no idea how it suppose to work.

I think `simplecss` is low-level enough that it's enough if it provides the information whether an attribute has the `important` value or not, which is what it currently does. And since style deduplication is handled on the `resvg` side right now, we have to do it there anyway.

File: crates/usvg/src/parser/svgtree/parse.rs
Comment (LaurenzV) [reply crates/usvg/src/parser/svgtree/parse.rs]: Don't think it's possible to avoid since we are overriding the value in some cases, but I created a shorthand variable now.

**Comment by RazrFalcon** [comment]
Looks good to me.

**Comment by RazrFalcon** [comment]
#252

**Comment by LaurenzV** [comment]
Will try it.

**Comment by LaurenzV** [comment]
Seems to work as intended. :)

Before:
test
sbahn2

After:
test
sbahn

**Comment by LaurenzV** [comment]
I just noticed this, though:

```
When a declaration is important, the order of precedence is reversed. Declarations marked as important in the user-agent style sheets override all important declarations in the user style sheets. Similarly, all important declarations in the user style sheets override all important declarations in the author's style sheets. Finally, all important declarations take precedence over all animations.
```

So I think I need to adapt the swapping condition a bit. Will do so a bit later.

**Comment by LaurenzV** [comment]
Should be good now.

**Comment by LaurenzV** [comment]
Is this okay to merge? Not sure who to ping, @waywardmonkeys maybe?

**Comment by waywardmonkeys** [comment]
@LaurenzV Well, I think @RazrFalcon said it looks good, so I think I can approve it and you can merge it?

**Comment by DJMcNab** [comment]
It's Linebender policy to not merge without an approval. https://github.com/linebender/resvg/pull/843#issuecomment-2466621536 is close enough to approval that I'm happy to formalise it, although I'll just give the actual change a quick scan.

**Review by waywardmonkeys — APPROVED** [review]

**Review by DJMcNab — APPROVED** [review]
Thanks!

**Comment by LaurenzV** [comment]
Thanks!

**Comment by DJMcNab** [comment]
> where I wanted to achieve that I can override all styles in the SVG via stylesheet injection

Incidentally, does the new swapping behaviour not break this? This injection is basically equivalent to a user agent/user stylesheet, right?

**Comment by DJMcNab** [comment]
Ooh, I see. That makes so much more sense.
I had the condition inverted in my head.

Because my intuition was that the user stylesheet would override the document, since that seems obviously correct. But that's not how the spec is written (the document overwriting the user agent makes sense of course, which I missed)

**Pull request merged**
